modules/packages/LinkedLists.chpl

```diff
@@ -410,40 +409,34 @@ record LinkedList {
     destroy();
 
     // Default format works as a 1D array
-    ref des = deserializer;
-    des.startArray(reader);
-    des.startArrayDim(reader);
+    var des = deserializer.startArray(reader);
+    des.startDim();
 
     var done = false;
     while !done {
       try {
-        append(des.readArrayElement(reader, eltType));
+        append(des.readElement(eltType));
```

Comment:
I noticed the `while des.hasMore()`  pattern was used for `list`, but not here. Is there a reason it can't be used in this case?

Reply:
This overload of ``deserialize`` exists for the DefaultDeserializer specifically because our default format technically wants to read/write LinkedLists as arrays. This overload exists for compatibility, and so it continues using the "Array"-specific part of the API.

We didn't include ``hasMore`` for arrays yet because of the general need to know array sizes ahead of time in Chapel. But it's something I think we can add post-2.0 if we want.